src/python/TaskWorker/Actions/SiteInfoResolver.py

```diff
@@ -0,0 +1,147 @@
+"""
+Resolve CRIC-related Sites info.
+"""
+# pylint: disable=too-many-branches
+
+import re
+import json
+from TaskWorker.DataObjects.Result import Result
+from TaskWorker.Actions.TaskAction import TaskAction
+from TaskWorker.WorkerExceptions import ConfigException, TaskWorkerException
+
+class SiteInfoResolver(TaskAction):
+    """
+    Given a task definition, resolve site white/black list with CRIC-related sites informations.
+    """
+
+    def __init__(self, config, crabserver, procnum=-1):
+        """ need a comment line here """
+        TaskAction.__init__(self, config, crabserver, procnum)
+        with config.TaskWorker.envForCMSWEB:
+            self.resourceCatalog = CRICService(logger=handler.logger, configDict={"cacheduration": 1, "pycurl": True, "usestalecache": True})
+
+    def isGlobalBlacklistIgnored(self, kwargs):
+        """ Determine wether the user wants to ignore the globalblacklist
+        """
+
+        return kwargs['task']['tm_ignore_global_blacklist'] == 'T'
+
```

Comment:
please add a (even dummy) comment in `"""..."""` to please `pylint`. I agree that the role of `execute` is rather obvious.